#Subject: [PATCH] revert stable write page patch series
#
#From: Zheng Liu <wenqing.lz@taobao.com>
#
#This patch reverts stable write page patch series which is introduced by Darrick
#J. Wong.  Google and Tao Bao met a latency problem after this patch series is
#applied.  When app does some unalignment writes and some dirty pages are being
#written out by flusher, app will wait on writeback.
#
#The patch series contains two patches:
#
#Darrick J. Wong(2):
#      mm: Wait for writeback when grabbing pages to begin a write (3d08bcc8)
#      fs: block_page_mkwrite should wait for writeback to finish (d76ee18a)
#
#Currently Darrick is working on improving stable write page feature.  So maybe
#at a proper time we can remove this patch.
#
#Patch-mainline: in-house
#Signed-off-by: Zheng Liu <wenqing.lz@taobao.com>
#---
# fs/buffer.c  |    1 -
# mm/filemap.c |    4 +---
# 2 files changed, 1 insertion(+), 4 deletions(-)

--- a/mm/filemap.c
+++ b/mm/filemap.c
@@ -2260,7 +2260,7 @@
 repeat:
 	page = find_lock_page(mapping, index);
 	if (page)
-		goto found;
+        return page;
 
 	page = __page_cache_alloc(gfp_mask & ~gfp_notmask);
 	if (!page)
@@ -2273,9 +2273,6 @@
 			goto repeat;
 		return NULL;
 	}
-found:
-	wait_on_page_writeback(page);
-	return page;
 }
 EXPORT_SYMBOL(grab_cache_page_write_begin);
 
